EPlast/EPlast.Tests/Controllers/UserControllerTests.cs

```diff
@@ -0,0 +1,610 @@
+            Assert.AreEqual(expectedListCount, (actual.ConfirmedUsers as List<ConfirmedUserViewModel>).Count);
+            Assert.AreEqual(expectedId, actual.ClubApprover.ID);
+            Assert.AreEqual(expectedId, actual.CityApprover.ID);
+            Assert.IsFalse(actual.IsUserHeadOfCity);
+            Assert.IsFalse(actual.IsUserHeadOfClub);
+            Assert.IsFalse(actual.IsUserHeadOfRegion);
+            Assert.IsFalse(actual.IsUserPlastun);
+            Assert.AreEqual(expectedApproverId, actual.CurrentUserId);
+        }
+
+        [Test]
+        public async Task ApproveUser_NullUserIdString_ReturnsNotFoundResult()
+        {
+            // Act
+            var result = await _userController.ApproveUser(It.IsAny<string>());
+
+            // Assert
+            _loggerService.Verify((x) => x.LogError(It.IsAny<string>()), Times.Once);
+            Assert.IsInstanceOf<NotFoundResult>(result);
+        }
+
+        [Test]
+        public async Task ApproveUser_ReturnsOkResult()
+        {
+            // Assert
+            var idString = "1";
+
+            _confirmedUserService
+                .Setup((x) => x.CreateAsync(It.IsAny<ClaimsPrincipal>(), idString, It.IsAny<bool>(), It.IsAny<bool>()));
+
+            // Act
+            var result = await _userController.ApproveUser(idString, It.IsAny<bool>(), It.IsAny<bool>());
+
+            // Assert
+            Assert.IsInstanceOf<OkResult>(result);
+        }
+
+        [Test]
+        public async Task ApproverDelete_NullUserIdString_ReturnsNotFoundResult()
+        {
+            // Act
+            var result = await _userController.ApproverDelete(It.IsAny<int>());
+
+            // Assert
+            _loggerService.Verify((x) => x.LogError(It.IsAny<string>()), Times.Once);
+            Assert.IsInstanceOf<NotFoundResult>(result);
+        }
+
+        [Test]
+        public async Task ApproverDelete_ReturnsOkResult()
+        {
+            // Assert
+            var confirmedId = 1;
+
+            _confirmedUserService
+                .Setup((x) => x.DeleteAsync(It.IsAny<int>()));
+
+            // Act
+            var result = await _userController.ApproverDelete(confirmedId);
+
+            // Assert
+            _loggerService.Verify((x) => x.LogInformation(It.IsAny<string>()), Times.Once);
+            Assert.IsInstanceOf<OkResult>(result);
+        }
+
+        public UserDTO CreateFakeUser()
+            => new UserDTO()
+            {
+                Id = "1",
+                FirstName = "SomeFirstName",
+                LastName = "SomeLastName",
+                UserProfile = new UserProfileDTO()
+                {
+                    EducationId = 1,
+                    WorkId = 1,
+                },
+            };
+
+        public UserViewModel CreateFakeUserViewModel()
+            => new UserViewModel()
+            {
+                ID = "1",
+            };
+
+        public EditUserViewModel CreateFakeEditUserViewModel()
```

Comment:
If you have faking function, use private modification.
```suggestion
        private EditUserViewModel CreateFakeEditUserViewModel()
```

Reply:
Done.